wallet/src/key_chain.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # BIP39 key chain
+//! The KeyChain struct holds and constantly derives keys for the wallet addresses
+//! It uses the following derivation scheme:
+//!
+//! m/0'/<account_number>'/<key_purpose>'/<key_index>'
```

Comment:
This is because we use hardened derivation for key_purpose and key_index so this does not comply with the standards. We might re-define this path to be compatible with the BIP44 standard.

Reply:
Some questions:
* Is there a plan to adopt the standard once soft derivations are supported?
* Would it make sense to adjust the path now to be more in sync with the standard? With the following modifications:
  * Not sure what to do with the initial `44'`. It's probably a good idea to change it to something else until the standard is followed correctly
  * Use something else for coin type, `0'` is bitcoin.
  * receiving/change and index components are hardened